### PR TITLE
DTSSTCI-86

### PR DESCRIPTION
**Pull request by tomxelliott**
### Change description ###
Create utility event for testing purposes that allows test case creation in any chosen state

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSSTCI-86

Screenshot 2023-12-22 at 15 53 37

Screenshot 2023-12-22 at 16 43 36

Screenshot 2023-12-22 at 16 43 28

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseData.java [hunk 1]
```diff
@@ -425,6 +426,14 @@ public class CaseData {
     @JsonUnwrapped
     private RetiredFields retiredFields;
 
+    @CCD(
+        label = "State to transition test case to",
+        typeOverride = FixedList,
+        typeParameterOverride = "State",
+        access = {DefaultAccess.class}
+    )
+    private State stateToTransitionApplicationTo;
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseData.java:435]: Is it possible to avoid introducing new fields for testing purpose only?

Comment (tomxelliott) [reply src/main/java/uk/gov/hmcts/sptribs/ciccase/model/CaseData.java]: Have updated using existing fields

**Review by gmmagruder — APPROVED** [review]

**Review by Jefex7 — DISMISSED** [review]

**Review by gmmagruder — DISMISSED** [review]

**Review by Jefex7 — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/sptribs/common/event/page/ContactPreferenceDetailsTest.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            .caseSubcategory(CaseSubcategory.SEXUAL_ABUSE)
+            .partiesCIC(Set.of(PartiesCIC.APPLICANT))
+            .build();
+        final CaseData caseData = CaseData.builder()
+            .cicCase(cicCase)
+            .build();
+        caseDetails.setData(caseData);
+
+        //When
+        final AboutToStartOrSubmitResponse<CaseData, State> response = contactPreferenceDetails.midEvent(caseDetails, caseDetails);
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/common/event/page/ContactPreferenceDetailsTest.java:40]: For consistency in the code, what is the preference with `final` modifiers?
Here on line 40 the response is `final`, but in the `CreateCaseTest`, several responses are not marked as final.

File: src/test/java/uk/gov/hmcts/sptribs/common/event/page/ContactPreferenceDetailsTest.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        //Then
+        assertThat(response.getErrors()).isNotNull();
```
Comment (oleon23) [inline src/test/java/uk/gov/hmcts/sptribs/common/event/page/ContactPreferenceDetailsTest.java:43]: Do we want to be more specific with why the Errors are not null? Are they an empty list (if everything was set-up correctly)? Or are there specific errors and messages we should be testing for?
e.g. `assertThat(response.getErrors()).contains(SELECT_AT_LEAST_ONE_ERROR_MESSAGE);`

**Review by oleon23 — DISMISSED** [review]
Looks good to me, no changes needed, my comments are more for discussion on what our conventions are in general as a team

**Review by Jefex7 — APPROVED** [review]

**Review by gmmagruder — APPROVED** [review]

**Pull request merged**